src/parser/on_demand/parse_to_elements.ts

```diff
@@ -0,0 +1,131 @@
+import { BSONOffsetError } from '../../error';
+import { NumberUtils } from '../../utils/number_utils';
+
+/**
+ * @public
+ * @experimental
+ */
+export type BSONElement = [
```

Comment:
why a tuple instead of an object?

Reply:
This was in the design. And there's a section in the Q&A about the thinking behind this. Something perhaps missing from the Q&A is how this function in end is in support of much more complex functions that determine how to parse the BSON, the intermediary results that this function produces can remain small.